tests/core/unittests/mongodb_test.py

```diff
@@ -47,10 +36,12 @@ def clean_db(database, exp_config):
 @pytest.fixture(scope='module')
 def moptdb():
     """Return MongoDB wrapper instance initiated with test opts."""
+    MongoDB.instance = None
     moptdb = MongoDB(username='user', password='pass', dbname='metaopt_test')
     return moptdb
 
 
+@pytest.mark.usefixtures("null_db_instances")
```

Comment:
last question, what is `null_db_instances`?

Reply:
It is a setup fixture that resets `Singleton`-type classes internal singleton instance object to None, so that we can test in the same run every possible invocation of a Singleton's creation. We assume independency between two different test functions.